scanner.cs

```diff
@@ -46,25 +47,29 @@ public class Scanner {
         }
     }
 
-    // This is the dispatcher
-    private int getNextToken() {
-        return 0; // change this
-    }
-
-    private int getLineNumber() {
-        return 0; // change this
-    }
-
-    private int getColumnNumber() {
-        return 0; // change this
+    private Token getNextToken() {
+        char next = __bytes[__curByte];
+        if(Constants.DIGITS.Contains("" + next)){
+            return fsaDigit();
+        } else if(Constants.PUNCTUATION.Contains("" + next)){
+            return fsaPunct();
+        } else if(Constants.LETTERS.Contains("" + next)){
```

Comment:
include '_' for simplicity since we're considering that a letter; BUT don't include the '_' in the actual LETTERS string, just as a second check.